ledger/accountdb.go

```diff
@@ -1293,6 +1297,7 @@ func (iterator *orderedAccountsIter) Next(ctx context.Context) (acct []accountAd
 		}
 
 		if err != nil {
+			iterator.Close(ctx)
 			return
 		}
 		iterator.step = oaiStepIterateOverOrderedTable
```

Comment:
Don't we need to close the iterator on line 1325?
```
			if len(addrbuf) != len(addr) {
				err = fmt.Errorf("Account DB address length mismatch: %d != %d", len(addrbuf), len(addr))
				return
			}
```

Reply:
yes, we do. I'll add that!